core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/samModels.scala

```diff
@@ -24,8 +24,12 @@ object SamResourceId {
     override def resourceType: SamResourceType = SamResourceType.Project
   }
 
-  final case class AppSamResourceId(resourceId: String) extends SamResourceId {
-    override def resourceType: SamResourceType = SamResourceType.App
+  final case class AppSamResourceId(resourceId: String, accessScope: Option[AppAccessScope]) extends SamResourceId {
+    override def resourceType: SamResourceType = accessScope match {
+      case Some(AppAccessScope.UserPrivate)     => SamResourceType.App
+      case Some(AppAccessScope.WorkspaceShared) => SamResourceType.SharedApp
+      case None                                 => SamResourceType.App
```

Comment:
This is where I chose to enforce the default value for the accessScope. My main reason for having this logic in here is because I did not want to carry along two different classes for the app and shared app. Doing this makes the rest a lot nicer.